tests/core/test_device_save_loader.py

```diff
@@ -73,6 +107,66 @@ async def test_enum_yaml_formatting(tmp_path):
     assert saved_enums == enums
 
 
+async def test_save_device_all_types(RE: RunEngine, device_all_types, tmp_path):
+    # Populate fake device with PV's...
+    await device_all_types.pv_int.set(1)
+    await device_all_types.pv_float.set(1.234)
+    await device_all_types.pv_str.set("test_string")
+    await device_all_types.pv_enum_str.set("two")
+    await device_all_types.pv_enum.set(MyEnum.two)
+    for pv, dtype in {
+        device_all_types.pv_array_int8: np.int8,
+        device_all_types.pv_array_uint8: np.uint8,
+        device_all_types.pv_array_int16: np.int16,
+        device_all_types.pv_array_uint16: np.uint16,
+        device_all_types.pv_array_int32: np.int32,
+        device_all_types.pv_array_uint32: np.uint32,
+        device_all_types.pv_array_int64: np.int64,
+        device_all_types.pv_array_uint64: np.uint64,
+    }.items():
+        await pv.set([np.iinfo(dtype).min, np.iinfo(dtype).max, 0, 1, 2, 3, 4])
```

Comment:
IIRC `MockSignalBackend` doesn't do any type conversion, so these should be numpy arrays rather than lists